feat(item): public methods

JohannesMerz opened the pull request:
new methods:

- item.update()
- item.delete()
- item.remove()
- item.fetch()

improvements/fixes:

- _transporterState is now passed to clientStorage.create/update so the state gets stored
- not found: if an item is not found in its destination it gets deleted

closes #7

Commit:
refactor(item): splits _synchronize into a more generic function

- _synchronizeFor can be called for a specific target
- POST_SYNC_PROCESSOR of transporter calls _synchronizeFor(clientStorage) instead
  of calling the methods directly. This makes sure that the clientStorage is
  only hit with valid data and when online.

Commit:
feat(item): remove, fetch, update.

implements follwoing methods:
  - remove
  - fetch
  - update

 Not found items results in item being removed.

Commit:
test(item): unit tests for delete, fetch and update

Commit:
fix(item): removing an not found item resolves proberly now.

Commit:
feat(item): includes transporter state in get for ClientStorage

Commit:
test(item): test for remove

JohannesMerz committed:
Merge branch 'master' into feature/item_public_methods

tchock commented on src/Item.js:
```diff
@@ -51,6 +52,63 @@ export default class Item {
+    return this.remove(source);
+  }
+
+  remove(source) {
```
shouldn't the source default value be moved from delete to remove?
This way users can also use remove without needing to add the source to the method.

JohannesMerz replied:
@tchock you are partial right. No default value is needed at all as its done by the switch case anyway. Removed it from item.delete. I also added a small fix in _synchronize (see the very last test in item.spec). Pls review again

tchock commented:
@JohannesMerz also: there is no test that covers the default value of remove/delete. When I removed the default value at all, the tests still all went green.
I guess these cases also should be added.

Commit:
fix(item): removes default source and fixes a bug in _synchronize

- removes default source from item.delete
- if an item comes back as not found and the initial action was to remove it,
  it now counts as removed and behaves as if the response came back resolved

Commit:
fix(item): remove sets state properly

JohannesMerz committed:
Merge branch 'master' into feature/item_public_methods
